### PR TITLE
server/asyncio.py change in the _log_exception method to correct the logging when disconnecting the socket

### PR DESCRIPTION
**Pull request by alexandrecuer**
in the _log_exception method :  
- all logging should be debug and not error
- no logging should be thrown if the handler is not running, otherwise cancellation messages are duplicated, we see the first cancellation message at the closing stage and the same is shown when we enter the closed state

this answers to https://github.com/riptideio/pymodbus/discussions/1110#discussioncomment-3871519

### CODE REVIEW

File: pymodbus/server/async_io.py [hunk 1]
```diff
@@ -100,11 +100,13 @@ def __init__(self, owner):
 
     def _log_exception(self):
         """Show log exception."""
+        if not self.running:
+            return
```
Comment (janiversen) [inline pymodbus/server/async_io.py:104]: Log exception can occur before and especially after server is running.

**Review by janiversen — CHANGES_REQUESTED** [review]

File: pymodbus/server/async_io.py (hunk 1 above)
Comment (alexandrecuer) [reply pymodbus/server/async_io.py:104]: The _log_exception() is only called twice, in handle() when self.running is True and in connection_lost(), so always after server is running. But maybe i missunderstand something

Comment (janiversen) [reply pymodbus/server/async_io.py:104]: _log_exception can be called from anywhere, we add/remove _log_exception often. Apart from that it is not logical nor correct that _log_exception depends on the state of the server.

**Review by janiversen — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**